src/index.rst

```diff
@@ -616,6 +647,59 @@ Now ready to submit the workflow: ::
 
    $> bps submit test_sdf.yaml
 
+Submit a development workflow to USDF
+-------------------------------------
+
+To submit a development workflow to S3DF, please at first check `Submit a workflow to USDF`_.
+Here we only list the differences.
+
+Copy the environment setup script from cvmfs to your local directory and update the lsst
+setup part to your private repo: ::
+
+   $> latest=$(ls -td /cvmfs/sw.lsst.eu/linux-x86_64/panda_env/v* | head -1)
+   $> cp $latest/setup_panda_s3df.sh .
+   $> <update setup_panda_s3df.sh>
+   $> source ./setup_panda_s3df.sh
+
+``Note``: Make sure PanDA can read your private repo: ::
+
+   $> chmod g+rxs <your private development repo>
+
+For the submission yaml file ``test_usdf.yaml``, you need to change the ``runnercommand`` to point to your private development repo: ::
+
+   $> cat test_usdf.yaml
+   # An example bps submission yaml
+   # Need to setup USDF before submitting the yaml
+   # source setupUSDF.sh
+
+   LSST_VERSION: w_2022_32
+
+   includeConfigs:
+   - ${CTRL_BPS_PANDA_DIR}/config/bps_usdf.yaml
+
+   pipelineYaml: "${DRP_PIPE_DIR}/pipelines/HSC/DRP-RC2.yaml#isr"
+
+   payload:
+     payloadName: testUSDF
+     inCollection: "HSC/RC2/defaults"
+     dataQuery: "exposure = 34342 AND detector = 10"
+
+   # To override the 'loadLSST.bash' and 'setup lsst_distrib' with your development repo.
+   runnerCommand: >
+      unset PYTHONPATH;
+      source /cvmfs/sw.lsst.eu/linux-x86_64/lsst_distrib/{LSST_VERSION}/loadLSST.bash;
+      pwd; ls -al;
+      # setup lsst_distrib;
+      setup -k -r /path/to/your/test/package;
```

Comment:
I am not sure how to find the information from 'loadLSST.sh' and 'setup -k -r' (since we need to setup this part too before 'bps submit'). If 'loadLSST.sh' and 'setup -k -r' has provided some environments, then we can add something in ctrl-bps-panda to find these environments to automatically set the runnercommand. Do you have any idea how to do it?

Reply:
This could be a long term question (e.g., something similar to HTCondor's getenv).  But for this I was meaning something like the following (which I haven't tested):

- In the user yaml:
`setupUserEnv: "setup -k -r /path/to/your/test/package; export SOME_OTHER_KEY_VAL='theval'"`

- In the plugin yaml:
```
setupEnv: "source /cvmfs/sw.lsst.eu/linux-x86_64/lsst_distrib/{LSST_VERSION}/loadLSST.bash; setup lsst_distrib; {setupUserEnv}"
runnerCommand:   >
  unset PYTHONPATH;
  {setupEnv};
  pwd; ls -al;
  python3 ${CTRL_BPS_PANDA_DIR}/python/lsst/ctrl/bps/panda/edgenode/cmd_line_decoder.py _cmd_line_ & pJob=$!;
  prmon -i 5
  -f memory_monitor_output.txt
  -j memory_monitor_summary.json
  -p $pJob & mJob=$!;
  wait $pJob;
  retStat=$?;
  wait $mJob;
  rm -fr EXEC_REPO-*;
  exit $retStat
```

That way you can change prmon command line, etc without everyone having to change their yaml.  They only have to set the "setupUserEnv"  (if an expert wants to source a different stack, they can override "setupEnv" and a PanDA expert can override "runnderCommand").  I haven't tested this to make sure the variables get substituted correctly.  But I think it's worth trying.

Reply:
I added a setupLSSTEnv, so that users can update it.
https://github.com/lsst/ctrl_bps_panda/pull/27

Reply:
My test tasks are also ok.